sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_mobile/HttpRequestMessageFactory.cs

```diff
@@ -425,8 +425,17 @@ public async System.Threading.Tasks.Task<IWebResponseData> GetResponseAsync(Syst
         {
             try
             {
-                var responseMessage = await _httpClient.SendAsync(_request, HttpCompletionOption.ResponseHeadersRead, cancellationToken)
-                    .ConfigureAwait(continueOnCapturedContext: false);
+                HttpResponseMessage responseMessage; 
+                try
+                {
+                    responseMessage = await _httpClient.SendAsync(_request, HttpCompletionOption.ResponseHeadersRead, cancellationToken)
+                        .ConfigureAwait(continueOnCapturedContext: false);
+                }
+                catch(ObjectDisposedException e)
```

Comment:
How can you be sure ObjectDisposedException is because of network connectivity failure?  Also, aren't we just re-throwing the exception?  Why not just rethrow ObjectDisposedException with better message instead?

Reply:
I am not sure ObjectDisposedException  is only due to a network connectivity failure, but we are not able to tell the difference of failure sources here, so I consider ObjectDisposedException is a kind of network connectivity failure. 
The reason I rethrow an IOException is because that our retry policy will handle IOException , but not ObjectDisposedException . We can rethrow ObjectDisposedException with a better message, but then we need add handler for this kind of exception in our retry policy. Which solution do you prefer to?